mantidproject/dockerfiles
# Tidy up development images

This does the following things:
- Default to a non-root user
- Ensures user environment is set correctly (including `USER` environment variable)
- Allows RW to host filesystem as your host user
- Includes ParaView `src` and `build` in more sensible lcoations

This fixes:
- Two failing unit tests
- Issues where you have trouble with root filesystem access (cc @soininen )

To test:
- Try the new images out (I have copies to test on [my Docker Hub account](https://hub.docker.com/u/dannixon)) and make sure they work. See the updated documentation, running them is slightly different now (you must pass in your host user and group IDs)
- See that once built, all unit tests pass inside the container
- See that build time, disk usage and kernel CPU usage are not overly affected
